lib/Jsrt/Core/JsrtCore.cpp

```diff
@@ -735,49 +735,23 @@ CHAKRA_API JsCloneObject(_In_ JsValueRef source, _Out_ JsValueRef* newObject)
     VALIDATE_JSREF(source);
 
     return ContextAPINoScriptWrapper([&](Js::ScriptContext* scriptContext, TTDRecorder& _actionEntryPopper) -> JsErrorCode {
-        if (Js::VarIs<Js::JavascriptProxy>(source))
+        Js::JavascriptProxy* proxy = Js::JavascriptOperators::TryFromVar<Js::JavascriptProxy>(source);
+        if (proxy != nullptr)
         {
-            source = Js::UnsafeVarTo<Js::JavascriptProxy>(source)->GetTarget();
-        }
-        JsrtExternalObject* externalSource = Js::JavascriptOperators::TryFromVar<JsrtExternalObject>(source);
-        if (externalSource)
-        {
-            JsrtExternalType* externalType = externalSource->GetExternalType();
-            JsrtExternalObject* target = JsrtExternalObject::Create(
-                externalSource->GetSlotData(),
-                externalSource->GetInlineSlotSize(),
-                externalType->GetJsTraceCallback(),
-                externalType->GetJsFinalizeCallback(),
-                externalSource->GetPrototype(),
-                scriptContext,
-                externalType);
-            bool success = target->TryCopy(externalSource, true);
-            AssertOrFailFast(success);
-            *newObject = target;
-            return JsNoError;
-        }
-        else
-        {
-            Js::CustomExternalWrapperObject * externalWrapper = Js::JavascriptOperators::TryFromVar<Js::CustomExternalWrapperObject>(source);
-            if (externalWrapper != nullptr) {
-                Js::CustomExternalWrapperObject * target = Js::CustomExternalWrapperObject::Clone(externalWrapper, scriptContext);
-                Assert(target);
-                *newObject = target;
-                return JsNoError;
-            }
+            source = proxy->GetTarget();
```

Comment:
what is the target itself is proxy?

Reply:
makes sense, we should follow the chain. i updated a couple other places in jsrt apis where we were doing the same